client/my-sites/upgrades/domain-management/components/header/index.jsx

```diff
@@ -11,7 +11,7 @@ const HeaderCake = require( 'components/header-cake' );
 const DomainManagementHeader = React.createClass( {
 	render() {
 		return (
-			<HeaderCake className="domain-management-header" onClick={ this.props.onClick }>
+			<HeaderCake className="domain-management-header" onClick={ this.props.onClick } backHref={ this.props.backHref }>
```

Comment:
Agreed, I don't know why `onClick` was preferred beforehand but `backHref` is much better.